tracing-flame/src/lib.rs

```diff
@@ -412,7 +410,11 @@ where
             stack += "all-threads";
         }
 
-        for parent in parents {
+        let mut parents = first.scope();
+        parents
+            .next()
+            .expect("expected: scope begins with leaf scope");
+        for parent in parents.from_root() {
             stack += "; ";
             write(&mut stack, parent, &self.config).expect("expected: write to String never fails");
         }
```

Comment:
Is it actually intentional here that the leaf scope is excluded?

Reply:
Also, even if it was, it'd probably be cleaner to replace this with `first.parent().into_iter().flat_map(|parent| parent.scope().from_root())` or similar..

Reply:
IIRC it is, because of how perf / flamegraph represent entering a stack frame. You have to have a previous frame, so we remove the current span to insert the "stack before this span was entered" entry.